packages/core/src/core.ts

```diff
@@ -651,18 +652,24 @@ class ThresholdKey implements ITKey {
   }: {
     determinedShare?: BN;
     initializeModules?: boolean;
-    importedKey?: BN;
+    importedKey?: Buffer;
     delete1OutOf1?: boolean;
   } = {}): Promise<InitializeNewKeyResult> {
-    let seed: Uint8Array = new Uint8Array();
+    let seed: Buffer;
     if (this.keyType === KeyType.secp256k1) {
-      const tmpPriv = importedKey || generatePrivate(this.keyType);
+      const tmpPriv = importedKey ? new BN(importedKey) : generatePrivate(this.keyType);
       this._setKey(new BN(tmpPriv));
     } else if (this.keyType === KeyType.ed25519) {
-      seed = importedKey ? new Uint8Array(importedKey.toBuffer(undefined, 32)) : nacl.randomBytes(32);
-      const keyPair = nacl.sign.keyPair.fromSeed(seed);
-      const tempPriv = new BN(keyPair.secretKey.slice(0, 32)).umod(this.ecCurve.curve.n);
-      this._setKey(tempPriv);
+      seed = importedKey && importedKey;
+      if (!seed) {
+        const newEd25519Seed = await getRandomBytes(32);
+        seed = Buffer.from(newEd25519Seed);
+      }
+      const keyPair = getEd25519ExtendedPublicKey(seed);
+      // note this scalar is litte endian encoded
+      // no need of converting to BE, since we only use this for encryption and decryption.
```

Comment:
It appears that this comment is incorrect.

The returned scalar is of type `BN`, therefore, it doesn't have an inherent le or be encoding. (This is only meaningful once you convert it to bytes, for example. `BN` supports both, be and le, btw.)

Reply:
yeah i think that correct, once it is converted to BN with right encoding, we dont have to care about endianess. I will remove this comment.